.github/workflows/tests.yaml

```diff
@@ -1,27 +1,229 @@
-name: PHP Composer
+name: Tests
 
 on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ master ]
+    push:
+        branches: [ master ]
+        paths-ignore:
+            - README.md
+    pull_request:
+        branches: [ master ]
+        paths-ignore:
+            - README.md
 
 jobs:
-  build:
 
-    runs-on: ubuntu-latest
+    php:
+        runs-on: ubuntu-latest
 
-    steps:
-    - uses: actions/checkout@v2
+        strategy:
+            fail-fast: false
+            matrix:
+                php: [7.2, 7.3, 7.4]
+                prefer: ["--prefer-lowest", ""]
 
-    - name: Validate composer.json and composer.lock
-      run: composer validate
+        env:
+            package-name: synolia/sylius-mail-tester-plugin
 
-    - name: Install dependencies
-      run: composer install --prefer-dist --no-progress --no-suggest
+        steps:
+            - uses: actions/checkout@v2
 
-    # Add a test script to composer.json, for instance: "test": "vendor/bin/phpunit"
-    # Docs: https://getcomposer.org/doc/articles/scripts.md
+            - name: PHP - Switch
+              run: sudo update-alternatives --set php /usr/bin/php${{ matrix.php }}
 
-    # - name: Run test suite
-    #   run: composer run-script test
+            - name: Composer - Get Cache Directory
+              id: composer-cache
+              run: echo "::set-output name=dir::$(composer config cache-files-dir)"
+
+            - uses: actions/cache@v1
+              id: cache-composer
+              with:
+                  path: ${{ steps.composer-cache.outputs.dir }}
+                  key: composer-php.${{ matrix.php }}-prefer.${{ matrix.prefer }}-${{ github.sha }}
+                  restore-keys: composer-php.${{ matrix.php }}-prefer.${{ matrix.prefer }}-
+
+            - name: Composer - Create cache directory
+              run: mkdir -p /home/runner/.composer/cache
+              if: steps.cache-composer.outputs.cache-hit != 'true'
+
+            - name: Composer - Self Update
+              run: |
+                  mkdir -p /home/runner/.composer/
+                  sudo composer self-update
+
+            - name: Composer - Validate composer.json and composer.lock
+              run: composer validate --strict
+
+            - name: Composer - Github Auth
+              run: composer config -g github-oauth.github.com ${{ github.token }}
+
+            - name: Composer - Update dependencies
+              run: composer update ${{ matrix.prefer }} --no-progress --no-scripts
+
+            - name: ECS - Run
+              run: (test -f ruleset/easy-coding-standard.yml && vendor/bin/ecs check src/ tests/Behat/ --no-progress-bar -c ruleset/easy-coding-standard.yml) || echo Ecs does not config, skipping step
+
+            - name: PHPStan - Run
+              run: (test -f ruleset/phpstan.neon && vendor/bin/phpstan analyse -c ruleset/phpstan.neon src/) || echo PHPStan does not config, skipping step
```

Comment:
|| echo PHPStan ruleset file does not exist

Reply:
Thanks.